python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -115,9 +126,59 @@ def get_nodes_to_launch(
         total_nodes_to_add = self._get_concurrent_resource_demand_to_launch(
             total_nodes_to_add, usage_by_ip.keys(), nodes, pending_nodes)
 
+        if self.is_legacy_yaml:
+            assert len(total_nodes_to_add.keys()) <= 1
         logger.info("Node requests: {}".format(total_nodes_to_add))
         return total_nodes_to_add
 
+    def _handle_legacy_yaml(self, nodes: List[NodeID],
+                            pending_nodes: Dict[NodeType, int],
+                            static_node_resources: List[ResourceDict]
+                            ) -> (bool, Dict[NodeType, int]):
+        """Handles legacy config files without available_node_types.
+
+        Launches a single node from which it calculates the resources and fills
```

Comment:
I do not understand your proposal, how it would calculate the resources?

Reply:
Well within `calculate_node_resources`, you know the node_type, and you know the ip address, so if you pass `load_metrics.static_resources` in, you can get the total resources for the node.

Reply:
How can you have an ip address if you haven't launched a node already? We only launch once after which we set the node types.

Reply:
Are you assuming that the type of the head node and worker nodes are the same?

Reply:
As an improvement, I can avoid this launch if the worker node is the same type as the head node.
Did you have something else in mind?

Reply:
You also have to give it a tag   ({None: count}) wont work as we use the multi node type node tags all over the place.

Reply:
> How can you have an ip address if you haven't launched a node already?

The node will still be launched if needed (by the legacy autoscaler right now). 

> Are you assuming that the type of the head node and worker nodes are the same?

No, but in this proposal, if the head and worker node types are the same, the worker's node resources will be detected with 0 launches already. 

> ({None: count}) wont work

Why not? None should be a perfectly valid key. If there's really some reason that we can't use `None` as a key, then it may be reasonable to have some constant. I imagine that in `prepare_config` you may need to do a `config.avaiable_node_types[None] = config["worker_nodes"]`

Reply:
so TLDR, you are suggesting to launch only if worker_nodes and head_node types are different.
The issue with {None: count} is that we have to check for many places where `if TAG_RAY_USER_NODE_TYPE in tags..`